src/main/java/net/openhft/chronicle/queue/util/PretouchUtil.java

```diff
@@ -18,16 +18,49 @@
 
 package net.openhft.chronicle.queue.util;
 
+import net.openhft.chronicle.core.Jvm;
 import net.openhft.chronicle.core.threads.EventHandler;
+import net.openhft.chronicle.core.util.ThrowingFunction;
+import net.openhft.chronicle.queue.impl.single.Pretoucher;
 import net.openhft.chronicle.queue.impl.single.SingleChronicleQueue;
-import net.openhft.chronicle.queue.internal.InternalPretouchHandler;
+import net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder;
 import org.jetbrains.annotations.NotNull;
 
+import java.lang.reflect.Method;
+import java.util.function.Function;
+
 public final class PretouchUtil {
+    private static final Function<SingleChronicleQueue, Pretoucher> PRETOUCHER_FACTORY;
+    private static final Function<SingleChronicleQueue, EventHandler> EH_FACTORY;
 
-    private PretouchUtil() {}
+    static {
+        Function<SingleChronicleQueue, Pretoucher> ptFactory;
+        Function<SingleChronicleQueue, EventHandler> ehFactory;
+        try {
+            final Class<?> clazz = Class.forName("software.chronicle.enterprise.queue.pretoucher.EnterprisePretouchUtil");
+            final Method createPretoucher = Jvm.getMethod(clazz, "createPretoucher", SingleChronicleQueue.class);
```

Comment:
Maybe we could convert Pretoucher to an interface and use it? Only cast it once instead of calling numerous reflected methods. That's fragile.

Reply:
Yes it is horrible, but I couldn't make Pretoucher an interface without breaking binary compatibility.

But - you know what - we said it was gonna be deprecated in x.24 so we *can* make it an interface - I'll do that, and break binary compatibility (that was my first solution but I got cold feet)